addons/mrp_repair/models/mrp_repair.py

```diff
@@ -0,0 +1,539 @@
+
+        self.address_id = addr['delivery'] or addr['default']
+        self.partner_invoice_id = addr['invoice']
+        self.pricelist_id = pricelist
+
+    @api.multi
+    def action_cancel_draft(self):
+        """ Cancels repair order when it is in 'Draft' state.
+        @param *arg: Arguments
+        @return: True
+        """
+        if not self:
+            return False
+        for repair in self:
+            repair.operations_ids.write({'state': 'draft'})
+        self.write({'state': 'draft'})
+        return self.create_workflow()
+
+    @api.multi
+    def action_confirm(self):
+        """ Repair order state is set to 'To be invoiced' when invoice method
+        is 'Before repair' else state becomes 'Confirmed'.
+        @param *arg: Arguments
+        @return: True
+        """
+        for order in self:
+            if (order.invoice_method == 'b4repair'):
+                self.write({'state': '2binvoiced'})
+            else:
+                self.write({'state': 'confirmed'})
+                for line in order.operations_ids:
+                    if line.product_id.tracking != 'none' and not line.lot_id:
+                        raise UserError(_("Serial number is required for operation line with product '%s'") % (line.product_id.name))
+                [line_order.write({'state': 'confirmed'}) for line_order in order.operations_ids]
+        return True
+
+    @api.multi
+    def action_cancel(self):
+        """ Cancels repair order.
+        @return: True
+        """
+        for repair in self:
+            if not repair.invoiced:
+                repair.operations_ids.write({'state': 'cancel'})
+            else:
+                raise UserError(_('Repair order is already invoiced.'))
+        return self.write({'state': 'cancel'})
+
+    def wkf_invoice_create(self):
+        self.action_invoice_create()
+        return True
+
+    @api.multi
+    def action_invoice_create(self, group=False):
+        """ Creates invoice(s) for repair order.
+        @param group: It is set to true when group invoice is to be generated.
+        @return: Invoice Ids.
+        """
+        res = {}
+        invoices_group = {}
+        AccountInvoiceLine = self.env['account.invoice.line']
+        AccountInvoice = self.env['account.invoice']
+        for repair in self:
+            res[repair.id] = False
+            if repair.state in ('draft', 'cancel') or repair.invoice_id:
+                continue
+            if not (repair.partner_id.id and repair.partner_invoice_id.id):
+                raise UserError(_('You have to select a Partner Invoice Address in the repair form!'))
+            comment = repair.quotation_notes
+            if (repair.invoice_method != 'none'):
+                if group and repair.partner_invoice_id.id in invoices_group:
+                    inv_id = invoices_group[repair.partner_invoice_id.id]
```

Comment:
change variable name. now invoices_group[repair.partner_invoice_id.id] contain recordset not id.